### PR TITLE
VAULT-893 Supervisord

### PR DESCRIPTION
**Pull request by Iapetus999**
You only need to look at commit abe715e
The rest is pulled from envoyproxy/ratelimit 
I did test this by sh into docker and killing the process. It came back.

### CODE REVIEW

**Comment by vmykh** [comment]
Can we split this pull request into two PRs?
1. Merge new changes from upstream (`envoyproxy/ratelimit`)
2. Add supervisord

Maybe it will be easier in future to distinguish between our changes and upstream changes

File: Dockerfile [hunk 1]
```diff
@@ -12,4 +12,11 @@ RUN CGO_ENABLED=0 GOOS=linux go build -o /go/bin/ratelimit -ldflags="-w -s" -v g
 
 FROM alpine:3.11 AS final
 RUN apk --no-cache add ca-certificates
+
+FROM ubuntu:latest as install
```
Comment (vmykh) [inline Dockerfile:16]: why do we need `as install` here

looks like this label is not used

**Review by vmykh — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (Iapetus999) [reply Dockerfile]: removed

**Comment by Iapetus999** [comment]
retest me please

**Pull request merged**